api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -356,4 +356,10 @@ private void validateInstitution(Institution institutionRequest) {
               });
     }
   }
+
+  public Optional<Institution> getFirstMatchingInstitution(final String contactEmail) {
+    return getInstitutions().stream()
```

Comment:
super nit: might be faster to do this via a query directly on the DAO, but with a small # of institutions it doesn't much matter.

Reply:
Good point – I'll avoid premature optimization in this case (especially since we do a full read of institutions in a few other places in this service).